File: MlbDataServer/DataStructures/Highlight.cs
using System;
using System.Xml.Serialization;

namespace MlbDataServer.DataStructures
{
	[Serializable, XmlRoot("media")]
	public class Highlight
	{
		[XmlAttribute("type")]
		public string Type { get; set; }

		[XmlAttribute("id")]
		public long Id { get; set; }

		[XmlAttribute("date")]
		public string Date { get; set; }

		[XmlElement("headline")]
		public string Headline { get; set; }

		[XmlElement("blurb")]
		public string Blurb { get; set; }

		[XmlElement("duration")]
		public string Duration { get; set; }

		[XmlElement("url")]
		public string[] Urls { get; set; }

		[XmlElement("thumb")]
		public string Thumb { get; set; }

		[XmlArray("thumbnails")]
		[XmlArrayItem("thumb")]
		public string[] Thumbs { get; set; }

		[XmlArray("keywords")]
		[XmlArrayItem("keyword")]
		public Keyword[] Keywords { get; set; }
	}

}

File: BaseballTheater/Areas/App/Scripts/Site/MlbDataServer/DataStructures/highlight.ts
// ReSharper disable InconsistentNaming
namespace Theater
{
	export interface IHighlight
	{
		type: string;
		id: number;
		team_id: string;
		date: Date;
		headline: string;
		blurb: string;
		bigblurb: string;
		duration: string;
		url: IUrl[];
		thumb: IUrl;
		thumbnails: IThumbnails;
		keywords: IKeywords;
		condensed: boolean;
		recap: boolean;
		"top-play": string;

		isPlaying: boolean;
	}
	
	export interface IHighlightSearchResult
	{
		Highlight: IHighlight;
		GameId: number;
	}

	export interface IThumbnails
	{
		thumb: IThumb[];
	}

	export interface IThumb
	{
		__text: string;

	}

	export interface IUrl
	{
		type: string;
		__text: string;
	}

	export interface IKeywords
	{
		keyword: Keyword[];
	}
}